storage-transfer/src/main/java/com/google/cloud/storage/storagetransfer/samples/TransferFromAzure.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2022 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage.storagetransfer.samples;
+
+// [START storagetransfer_transfer_from_azure]
+import com.google.storagetransfer.v1.proto.StorageTransferServiceClient;
+import com.google.storagetransfer.v1.proto.TransferProto;
+import com.google.storagetransfer.v1.proto.TransferProto.RunTransferJobRequest;
+import com.google.storagetransfer.v1.proto.TransferTypes.AzureBlobStorageData;
+import com.google.storagetransfer.v1.proto.TransferTypes.AzureCredentials;
+import com.google.storagetransfer.v1.proto.TransferTypes.GcsData;
+import com.google.storagetransfer.v1.proto.TransferTypes.TransferJob;
+import com.google.storagetransfer.v1.proto.TransferTypes.TransferJob.Status;
+import com.google.storagetransfer.v1.proto.TransferTypes.TransferSpec;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+public class TransferFromAzure {
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Your Google Cloud Project ID
+    String projectId = "my-project-id";
+
+    // Your Azure Storage Account name
+    String azureStorageAccountName = "myAzureAccountName";
+
+    // The Azure source container to transfer data from
+    String azureSourceContainer = "my-source-container";
+
+    // The GCS bucket to transfer data to
+    String gcsSinkBukcet = "my-sink-bucket";
+
+    transferFromAzureBlobStorage(
+        projectId, azureStorageAccountName, azureSourceContainer, gcsSinkBukcet);
+  }
+
+  public static void transferFromAzureBlobStorage(String projectId, String azureStorageAccountName,
+      String azureSourceContainer, String gcsSinkBucket)
+      throws IOException, ExecutionException, InterruptedException {
+
+    // Your Azure SAS token, should be accessed via environment variable
+    String azureSasToken = System.getenv("AZURE_SAS_TOKEN");
```

Comment:
Should the azureSasToken be obtained only as an env variable? Is this a best practice to be followed?
If not, can we obtain it from the main method?

Reply:
It in fact needs to be only obtained through env variable. If it's passed in through the method, if a test fails, it can end up leaking the credential in the test logs. We've run into this issue in the past and as a result we need sensitive data to be accessed through environment variables. It's also a safer practice in general